libp2p/protocols/perf/client.nim

```diff
@@ -57,7 +59,13 @@ proc perf*(
       statsCopy.uploadBytes += toWrite.uint
       p.stats = statsCopy
 
-    await conn.close()
+    # Close connection after writing for TCP, but not for QUIC
+    when defined(libp2p_quic_support):
+      if not (conn of QuicStream):
+        await conn.close()
+      # For QUIC streams, don't close yet - let server manage lifecycle
+    else:
+      await conn.close()
```

Comment:
this would be nicer if could be written like this
```
 if not (conn of QuicStream):
        await conn.close()
```

without  `when defined(libp2p_quic_support):` because if it is quic conn then we can assume that quic is supported. and in other places...

Reply:
is `when defined(libp2p_quic_support)` necessary in these cases?